src/migration/MigrationExecutor.ts

```diff
@@ -48,6 +48,40 @@ export class MigrationExecutor {
     // Public Methods
     // -------------------------------------------------------------------------
 
+    public async executeMigration(migration: Migration): Promise<Migration> {
+        return this.withQueryRunner(async (queryRunner) => {
+            await this.createMigrationsTableIfNotExist(queryRunner);
+            await (migration.instance as any).up(queryRunner);
+            await this.insertExecutedMigration(queryRunner, migration);
+
+            return migration;
+        })
+    }
+
+    public async getAllMigrations(): Promise<Migration[]> {
```

Comment:
why do we need it if we already have getMigrations ?

Reply:
getMigrations is a protected method where this one is part of the public API proposed returning a Promise<Mutation[]>, like its sibling methods getExecutedMigrations and getPendingMigrations.

Reply:
not sure about it. Not sure we really need this "all" name. But we can left it as you did.

Reply:
I just thought that for consistency they should all return promises instead and I didn't want to introduce any changes to current behaviour. I'll change getMigrations to return a Promise and make it public, if that's ok.

Reply:
choose any approach you prefer